### PR TITLE
Handle the loading of causal models more gracefully

### PR DESCRIPTION
**Pull request by gaugup**
## Description

It seems like loading of pickled causal models can fail and crash the `RAIInsights.load()` call. Hence, adding `try`/`except` logic when loading the pickled causal models.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] Documentation was updated if it was needed.
- [x] New tests were added or changes were manually verified.

### CODE REVIEW

File: responsibleai/responsibleai/_tools/shared/attribute_serialization.py [hunk 1]
```diff
@@ -113,8 +113,13 @@ def _load_attribute(
...
 ) -> Any:
     if file_format == SerializationFormats.PICKLE:
-        with open(path, 'rb') as f:
-            return pickle.load(f)
+        val = None
+        try:
+            with open(path, 'rb') as f:
+                val = pickle.load(f)
+        except Exception:
+            pass
```
Comment (riedgar-ms) [inline responsibleai/responsibleai/_tools/shared/attribute_serialization.py:121]: Should we at least write out the exception as a warning?

Comment (gaugup) [reply responsibleai/responsibleai/_tools/shared/attribute_serialization.py]: Done

File: responsibleai/tests/causal/test_causal_manager.py [hunk 2]
```diff
@@ -48,6 +51,30 @@ def test_causal_save_and_load(self, housing_data, tmpdir):
...
+        # models.
+        all_causal_dirs = DirectoryManager.list_sub_directories(save_dir)
+        for causal_dir in all_causal_dirs:
+            dm = DirectoryManager(parent_directory_path=save_dir,
+                                  sub_directory_name=causal_dir)
+            causal_analysis_pkl_file_path = \
+                dm.get_data_directory() / "causal_analysis.pkl"
+            os.remove(causal_analysis_pkl_file_path)
+
+        model_load_err = ('ERROR-LOADING-EXPLAINER: '
```
Comment (riedgar-ms) [inline responsibleai/tests/causal/test_causal_manager.py:65]: Error loading explainer in a causal test?

File: responsibleai/tests/causal/test_causal_manager.py [hunk 3]
```diff
@@ -48,6 +51,30 @@ def test_causal_save_and_load(self, housing_data, tmpdir):
...
+                          'There was an error loading the explainer. '
+                          'Some of RAI dashboard features may not work.')
+        with pytest.warns(UserWarning, match=model_load_err):
```
Comment (riedgar-ms) [inline responsibleai/tests/causal/test_causal_manager.py:68]: Thanks for checking the warning message as well.

File: responsibleai/responsibleai/_tools/shared/attribute_serialization.py [hunk 4]
```diff
@@ -113,8 +114,16 @@ def _load_attribute(
...
 ) -> Any:
     if file_format == SerializationFormats.PICKLE:
-        with open(path, 'rb') as f:
-            return pickle.load(f)
+        val = None
+        try:
+            with open(path, 'rb') as f:
+                val = pickle.load(f)
+        except Exception:
+            model_load_err = ('ERROR-LOADING-EXPLAINER: '
```
Comment (riedgar-ms) [inline responsibleai/responsibleai/_tools/shared/attribute_serialization.py:122]: Explainer?

File: responsibleai/tests/causal/test_causal_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/tests/causal/test_causal_manager.py:65]: Explainer is generic term. The warning is being raised from some generic code so chose explainer.

File: responsibleai/tests/causal/test_causal_manager.py [hunk 5]
```diff
@@ -48,6 +51,25 @@ def test_causal_save_and_load(self, housing_data, tmpdir):
         assert post_result.local_effects is not None
         assert post_result.policies is not None
 
+        # Remove the causal analaysis models to test the loading of
```
[the remark this answers is not in the document]
Comment (gaugup) [reply responsibleai/tests/causal/test_causal_manager.py]: done

**Pull request merged**